controllers/datadogagent/controller_reconcile_agent.go

```diff
@@ -316,8 +318,24 @@ func (r *Reconciler) cleanupDaemonSetsForProvidersThatNoLongerApply(ctx context.
 			r.log.Info("Deleted DaemonSet", "daemonSet.Namespace", ds.Namespace, "daemonSet.Name", ds.Name)
 			event := buildEventInfo(ds.Name, ds.Namespace, daemonSetKind, datadog.DeletionEvent)
 			r.recordEvent(dda, event)
+
+			removeDaemonSetStatus(ddaStatus, ds.Name)
 		}
 	}
 
 	return nil
 }
+
+// removeDaemonSetStatus removes a DaemonSet's status from a DatadogAgent's
+// status based on the DaemonSet's name
+func removeDaemonSetStatus(ddaStatus *datadoghqv2alpha1.DatadogAgentStatus, name string) {
```

Comment:
since this is used to clean up the EDS too, could we use a func name that is agnostic?
Like `removeStaleStatus` or something else?